File: OnlineShop.Web/src/app/models/item.ts
export class Item {
  id : number = 0;
  name: string = "";
  description: string = "";
  volume: string | null = null;
  price: number | null = 0;
  minimalBuyQuantity: number | null = 0;
  quantityInStock: number | null = 0;
  color: ItemColor | null = null;
  hasPhoto: boolean = false;

  constructor() {
    return {
      id: 0,
      name: "",
      description: "",
      volume: null,
      price: 0,
      minimalBuyQuantity: 0,
      quantityInStock: 0,
      color: null,
      hasPhoto: false
    }
  }
}

export class ItemColor {
  id: number = 0;
  name: string = "";
  colorHex: string = "";
}

File: OnlineShop.Api/OnlineShop.Domain/Item.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OnlineShop.Domain
{
    public class Item
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(50)]
        public string Name { get; set; }
        
        public string Description { get; set; }
        
        public string? ImageFileName { get; set; }
        
        [Column(TypeName = "decimal(9,2)")]
        public decimal Price { get; set; }
        
        [Range(1, 1_000_000)]
        public uint MinimalBuyQuantity { get; set; }
        
        [Range(1, 1_000_000)]
        public uint QuantityInStock { get; set; }
        
        [Range(1, 1_000_000)]
        public uint? Volume { get; set; }

        public int? ColorId { get; set; }
        
        public ItemColor? Color { get; set; }
    }
}